cmd/poseidon/main.go

```diff
@@ -112,18 +112,19 @@ func initProfiling(options config.Profiling) (cancel func()) {
 }
 
 // watchMemoryAndAlert monitors the memory usage of Poseidon and sends an alert if it exceeds a threshold.
-func watchMemoryAndAlert() {
-	// We assume that Poseidon usually takes about 50-300 MB of memory. Therefore, we specify the threshold of 1 GB.
-	// Improve: Make this value dynamic or relative.
-	const threshold = 1 * 1000 * 1000 * 1000
-	const interval = 5 * time.Second
+func watchMemoryAndAlert(options config.Profiling) {
+	if options.MemoryInterval == 0 {
+		return
+	}
 
+	var exceeded bool
```

Comment:
When is the value of this variable set? I see two usages below, but none of them changes the value. Am I missing something?

Reply:
No, I missed adjusting its value. Thank you for catching that!

Reply:
Ah, okay. Now, it makes sense. You remember whether the last check already saw an exceeded memory use and each time this observation changes, you issue a corresponding message.